great_expectations/util.py

```diff
@@ -865,16 +873,55 @@ def filter_properties_dict(
             [key for key, value in properties.items() if key in delete_fields]
         )
 
-    if clean_empty:
+    if clean_nulls:
+        keys_for_deletion.extend(
+            [
+                key
+                for key, value in properties.items()
+                if not (
+                    (keep_fields and key in keep_fields)
+                    or (delete_fields and key in delete_fields)
+                    or value is not None
+                )
+            ]
+        )
+
+    if clean_falsy:
+        if keep_falsy_numerics:
+            keys_for_deletion.extend(
+                [
+                    key
+                    for key, value in properties.items()
+                    if not (
+                        (keep_fields and key in keep_fields)
+                        or (delete_fields and key in delete_fields)
+                        or is_numeric(value=value)
+                        or value
+                    )
+                ]
+            )
+        else:
+            keys_for_deletion.extend(
+                [
+                    key
+                    for key, value in properties.items()
+                    if not (
+                        (keep_fields and key in keep_fields)
+                        or (delete_fields and key in delete_fields)
+                        or value
+                    )
+                ]
+            )
+
+    if clean_nulls and not clean_falsy:
         keys_for_deletion.extend(
             [
                 key
                 for key, value in properties.items()
                 if not (
                     (keep_fields and key in keep_fields)
                     or (delete_fields and key in delete_fields)
-                    or is_numeric(value=value)
-                    or value
+                    or value is not None
                 )
             ]
         )
```

Comment:
It looks like this is already taken care of on line 876, or am I missing something?

Reply:
The fix was committed.